sonar-db/src/main/java/org/sonar/db/permission/PermissionDao.java

```diff
@@ -19,166 +19,97 @@
  */
 package org.sonar.db.permission;
 
-import java.util.HashMap;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import javax.annotation.Nullable;
-import org.apache.ibatis.session.ResultHandler;
-import org.apache.ibatis.session.RowBounds;
-import org.apache.ibatis.session.SqlSession;
-import org.sonar.api.security.DefaultGroups;
 import org.sonar.db.Dao;
 import org.sonar.db.DbSession;
 import org.sonar.db.MyBatis;
-import org.sonar.db.user.GroupRoleDto;
-import org.sonar.db.user.UserPermissionDto;
 
 import static com.google.common.collect.Maps.newHashMap;
 import static org.sonar.db.DatabaseUtils.executeLargeInputs;
-import static org.sonar.db.DatabaseUtils.executeLargeInputsWithoutOutput;
 
+/**
+ * Only the operations involving both user and group permissions.
+ *
+ * @see GroupPermissionDao
+ * @see UserPermissionDao
+ */
 public class PermissionDao implements Dao {
 
-  private static final String QUERY_PARAMETER = "query";
-  private static final String COMPONENT_ID_PARAMETER = "componentId";
-  private static final String ANYONE_GROUP_PARAMETER = "anyoneGroup";
-
-  private final MyBatis myBatis;
+  private static final String USER_ID_PARAM = "userId";
+  private final MyBatis mybatis;
```

Comment:
I didn't refactor PermissionDao and PermissionMapper in this PR. I only dropped code.